src/sentry/snuba/subscriptions.py

```diff
@@ -156,12 +168,19 @@ def create_snuba_subscription(
     :param snuba_query: A `SnubaQuery` instance to subscribe the project to.
     :return: The QuerySubscription representing the subscription
     """
+    timebox_start = None
+    timebox_end = None
+    if timebox:
+        timebox_start = timezone.now()
+        timebox_end = timebox_start + timedelta(seconds=snuba_query.time_window)
```

Comment:
This logic shouldn't live here, we should pass these fields into the function and have your alert type handle the logic